File: data/gpt-4o/readable_format_HPEC/fortran/file197.f90
subroutine init_array(n, alpha, beta, a, b, x)
implicit none

DATA_TYPE, dimension(n, n) :: a
DATA_TYPE, dimension(n, n) :: b
DATA_TYPE, dimension(n) :: x
DATA_TYPE :: alpha, beta
integer :: n
integer :: i, j

alpha = 43532.0D0
beta = 12313.0D0

do i = 1, n
x(i) = DBLE(i - 1) / DBLE(n)
do j = 1, n
a(j, i) = ((DBLE(i - 1) * DBLE(j - 1))) / DBLE(n)
b(j, i) = ((DBLE(i - 1) * DBLE(j - 1))) / DBLE(n)
end do
end do
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file197.cpp
#include <vector>

void init_array(int n, double& alpha, double& beta, std::vector<std::vector<double>>& a, std::vector<std::vector<double>>& b, std::vector<double>& x)
{
 alpha = 43532.0;
 beta = 12313.0;

 for (int i = 0; i < n; i++)
 {
  x[i] = static_cast<double>(i) / static_cast<double>(n);
  for (int j = 0; j < n; j++)
  {
   a[j][i] = static_cast<double>(i * j) / static_cast<double>(n);
   b[j][i] = static_cast<double>(i * j) / static_cast<double>(n);
  }
 }
}
